react-components/src/architecture/base/renderTarget/UnitSystem.ts

```diff
@@ -61,7 +62,12 @@ export class UnitSystem {
   }
 
   public toStringWithUnit(value: number, quantity: Quantity): string {
-    return `${this.toString(value, quantity)} ${this.getUnit(quantity)}`;
+    const text = this.toString(value, quantity);
+    const unit = this.getUnit(quantity);
+    if (isEmpty(unit)) {
+      return text;
+    }
+    return `${text} ${unit}`;
```

Comment:
Do we need to handle the empty case explicitly? Or is this to handle cases where `unit` is e.g. `undefined`?